lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -203,7 +207,57 @@ namespace Js
             return instance;
         }
     };
-#endif
+
+#elif INTL_ICU
+
+    template<typename T>
+    class AutoIcuJsObject : public FinalizableObject
+    {
+    private:
+        T *instance;
+
+    public:
+        DEFINE_VTABLE_CTOR_NOBASE(AutoIcuJsObject<T>);
+
+        AutoIcuJsObject(T *object)
+            : instance(object)
+        { }
+
+        static AutoIcuJsObject<T> * New(Recycler *recycler, T *object)
+        {
+            return RecyclerNewFinalized(recycler, AutoIcuJsObject<T>, object);
+        }
+
+        void Finalize(bool isShutdown) override
+        {
+        }
+
+        void Dispose(bool isShutdown) override
+        {
+            if (!isShutdown)
+            {
+                // Here we use Cleanup() because we can't rely on delete (not dealing with virtual destructors).
+                // The template thus requires that the type implement the Cleanup function.
+                instance->Cleanup(); // e.g. deletes the object held in the IPlatformAgnosticResource
+
+                // REVIEW (doilij): Is cleanup in this way necessary or are the trivial destructors enough, assuming Cleanup() has been called?
+                // Note: delete here introduces a build break on Linux complaining of non-virtual dtor
+                // delete instance; // deletes the instance itself
+                // instance = nullptr;
+            }
+        }
+
+        void Mark(Recycler *recycler) override
+        {
+        }
+
+        T * GetInstance()
+        {
+            return instance;
+        }
+    };
+
+#endif // Defining Finalizable wrappers for Intl data
```

Comment:
I think this makes some sense as-is since the concept of this block is wrapper objects, but this is a strange way to comment an `#endif`.

Consider the perspective of someone unfamiliar with the code and reading through preprocessor config, `#endif // INTL_ICU` is more helpful to see which `#ifdef` or `#elif` it matches when you're reading up to find the beginning of the block, since preprocessor directives are not indented and "navigate to beginning of block" is still a little-supported feature of IDEs, especially for preprocessor blocks.

Documentation of this block is already given in the comment at the beginning, so that is sufficient.